static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -0,0 +1,190 @@
+
+  const [selectedOptions, setselectedOptions] = useState<string[]>(
+    dashboard.permissions?.isCreatorOnlyEditable
+      ? ['_creator']
+      : ['_everyone', '_creator', ...teamIds]
+  );
+
+  let isEverythingSelected =
+    selectedOptions.length === ['_everyone', '_creator', ...teamIds].length;
+
+  // Dashboard creator option in the dropdown
+  const makeCreatorOption = () => ({
+    value: '_creator',
+    label: (
+      <UserBadge
+        avatarSize={18}
+        user={dashboardCreator}
+        displayName={
+          <StyledDisplayName>
+            {dashboardCreator?.id === currentUser.id
+              ? tct('You ([email])', {email: currentUser.email})
+              : dashboardCreator?.email || currentUser.email}
+          </StyledDisplayName>
+        }
+        displayEmail={t('Creator')}
+      />
+    ),
+    textValue: '_creatorbadge',
+    disabled: dashboardCreator?.id !== currentUser.id,
+    // Creator option is always disabled
+    leadingItems: <Checkbox size="sm" checked disabled />,
+    hideCheck: true,
+  });
+
+  // Single team option in the dropdown [WIP]
+  const makeTeamOption = (team: Team) => ({
+    value: team.id,
+    label: `#${team.slug}`,
+    leadingItems: <TeamAvatar team={team} size={18} />,
+  });
+
+  // Avatars/Badges in the Edit Selector Button
+  const triggerAvatars =
+    isEverythingSelected || !dashboardCreator ? (
+      <StyledBadge key="_all" text={'All'} />
+    ) : (
+      <StyledAvatarList key="avatar-list" users={[dashboardCreator]} avatarSize={25} />
+    );
+
+  const dropdownOptions = [
+    makeCreatorOption(),
+    {
+      value: '_everyone_section',
+      options: [
+        {
+          value: '_everyone',
+          label: t('Everyone'),
+          disabled: dashboardCreator?.id !== currentUser.id,
+        },
+      ],
+    },
+    // [WIP: Selective edit access to teams]
+    {
+      value: '_teams',
+      label: t('Teams'),
+      options: teams.map(makeTeamOption),
+      showToggleAllButton: true,
+      disabled: true,
+    },
+  ];
+
+  // Handles state change when dropdown options are selected
+  const onSelectOptions = newSelectedOptions => {
+    const newSelectedValues = newSelectedOptions.map(
+      (option: {value: string}) => option.value
+    );
+    isEverythingSelected = false;
+    if (newSelectedValues.includes('_everyone')) {
+      isEverythingSelected = true;
+      setselectedOptions(['_everyone', '_creator', ...teamIds]);
+      if (newDashboardPermissions) {
+        newDashboardPermissions.isCreatorOnlyEditable = false;
+      }
+    } else if (!newSelectedValues.includes('_everyone')) {
+      setselectedOptions(['_creator']);
+      if (newDashboardPermissions === undefined) {
+        // When the dashboard does not have a permissions model associated
+        setNewDashboardPermissions({isCreatorOnlyEditable: true});
+      } else {
+        newDashboardPermissions.isCreatorOnlyEditable = true;
+      }
+    }
+  };
```

Comment:
Do we need to store the newDashboardPermissions every time a change happens? Can we just store the selected options and then when we trigger `onClose` we can create the new dashboard permissions for the request adhoc?

That way we can get rid of a state variable for `newDashboardPermissions`

Reply:
Removed the state hook for `newDashboardPermissions`, and added [this function](https://github.com/getsentry/sentry/blob/48a638d9ced60c8a02e4d09a6fcb62292e8e94b3/static/app/views/dashboards/editAccessSelector.tsx#L126)